server/sonar-search/src/main/java/org/sonar/search/SearchServer.java

```diff
@@ -50,56 +57,177 @@ public SearchServer(Props props) {
+            latch.countDown();
+          }
+        }
+      } catch (IOException e) {
+        e.printStackTrace();
+      }
+    }).start();
+
+    try {
+      latch.await();
+    } catch (InterruptedException e) {
+      // no action required
+    }
+
+    String urlString = url+"/_cluster/health?wait_for_status=yellow&timeout=30s";
+    try {
+      URL url = new URL(urlString);
+      url.openConnection();
+    } catch (MalformedURLException e) {
+      e.printStackTrace();
+    } catch (IOException e) {
+      e.printStackTrace();
     }
-    node = NodeBuilder.nodeBuilder().settings(esSettings).build();
-    node.start();
   }
 
-  // copied from https://github.com/elastic/elasticsearch/blob/v2.3.3/core/src/main/java/org/elasticsearch/bootstrap/Bootstrap.java
-  private static void initBootstrap() {
-    // init lucene random seed. it will use /dev/urandom where available:
-    StringHelper.randomId();
+  private Path findExecutable() {
+    String executable = getExecutable();
+    Path path = Paths.get(executable);
+    if (path.toFile().exists()) {
+      return path.toAbsolutePath();
+    }
+
+    String assembly = "sonar-application/src/main/assembly/";
+    Path idePath = Paths.get(assembly + executable);
+    if (idePath.toFile().exists()) {
+      return idePath.toAbsolutePath();
+    }
+
+    String parentAssembly = "../" + assembly;
+    Path buildServerPath = Paths.get(parentAssembly + executable);
+    if (buildServerPath.toFile().exists()) {
+      return buildServerPath.toAbsolutePath();
+    }
+
+    String grandParentAssembly = "../" + parentAssembly;
+    Path buildServerPath2 = Paths.get(grandParentAssembly + executable);
+    if (buildServerPath2.toFile().exists()) {
+      return buildServerPath2.toAbsolutePath();
+    }
+
+    throw new IllegalStateException(String.format("Cannot find elasticsearch binary %s in either %s or %s or %s or %s",
+            executable,
+            Paths.get(".").toAbsolutePath().getParent(),
+            Paths.get(assembly).toAbsolutePath(),
+            Paths.get(parentAssembly).toAbsolutePath(),
+            Paths.get(grandParentAssembly).toAbsolutePath()
+    ));
+  }
+
+  private static String getExecutable() {
+    if (System.getProperty("os.name").startsWith("Windows")) {
+      return "elasticsearch/bin/elasticsearch.bat";
+    }
+    return "elasticsearch/bin/elasticsearch";
   }
 
   @Override
   public Status getStatus() {
-    boolean esStatus = node != null && node.client().admin().cluster().prepareHealth()
-        .setWaitForYellowStatus()
-        .setTimeout(TimeValue.timeValueSeconds(30L))
-        .get()
-        .getStatus() != ClusterHealthStatus.RED;
-    if (esStatus) {
-      return Status.OPERATIONAL;
+    Status status = null;
+    try {
+      status = getStatus2();
+      System.out.println("ES STATUS "+status);
+      return status;
+    } catch (Exception e) {
+      System.out.println("ES STATUS "+e.getMessage());
+      throw new RuntimeException(e);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS00112)